hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4542,8 +4541,6 @@ func checkAdvertiseAddressOverlapping(hc *hyperv1.HostedCluster) field.ErrorList
 		return errs
 	}
 
-	advAddress = netip.MustParseAddr(advAddr.String())
```

Comment:
@enxebre To answer your question above about the change in findAdvertiseAddress, it was to remove the requirement to convert the output a second time with MustParseAddr() here. The code actually wants a netip.Addr, so it's simpler to return that in the first instance.